src/base/rss/rss_autodownloader.cpp

```diff
@@ -245,6 +245,7 @@ void AutoDownloader::processJob(const QSharedPointer<ProcessingJob> &job)
         }
 
         rule.setLastMatch(job->articleDate);
+        rule.matched(job->articleTitle);
```

Comment:
>And it's a lot of effort to go to, just to avoid a duplicate call to computeEpisodeName(), which only happens when there is a match.

Yes. This "duplicate call" is maybe not so bad in itself, but it indicates wrong logic in the code.
>That method seems like it would be very easy for lastEpisodeName to be leaked at some point. 

Agree. It was not a good example. I thought about it. Then we just need to achieve a consistent interface and behavior. We just need to do some things explicitly:
1. keeping last matched episode name is correct behavior
2. get last matched episode name is correct interface (like in regular expressions, we can get some matched data)
3. having `appendPreviouslyMatchedEpisode()` is good idea (and correct interface)
We get the following:
```c++
// autodownloadrule.cpp
void AutoDownloadRule::appendPreviouslyMatchedEpisode(const QString &episodeName)
{
    if (!episodeName.isEmpty() && !m_dataPtr->previouslyMatchedEpisodes.contains(episodeName))
        m_dataPtr->previouslyMatchedEpisodes.append(episodeName);
}

// autodownloader.cpp
rule.setLastMatch(job->articleDate);
rule.appendPreviouslyMatchedEpisode(rule.matchedEpisode());
```